### PR TITLE
Readme and dockerfile

### LINKED ISSUE #31: Building in docker

### PR DESCRIPTION
**Pull request by NickVolynkin**
### docker: reduce image size, reformat Dockerfile

* Build in less steps, remove caches and temporary files
  to reduce image size.
* Reformat Dockerfile for better readability and diffability.
* Add missing dependency `automake`.

### docker: build and lint Dockerfile in CI

* Build Dockerfile to check its correctness.
* Lint Dockerfile with Hadolint. Read the docs at
  https://github.com/hadolint/hadolint#readme

### docs: proofread the README.md

---

Image sizes:
```
docker image ls | grep toolchain                                                                                                                   
proof-market-toolchain         latest                4fe272424ddb   2 minutes ago   2.05GB
proof-market-toolchain         before                3afc294f98eb   24 hours ago    2.6GB
```

Readme preview at https://github.com/NilFoundation/proof-market-toolchain/tree/31-proofread-readme#readme

Resolves #31

### CODE REVIEW

**Comment by NickVolynkin** [comment]
I just understood that readme should be restructured. There's no need to go all the way with dependencies in the containerized builds. Will do it in a next issue #38.

File: README.md [hunk 1]
```diff
@@ -61,71 +62,143 @@ ragel >= 6.10
...
+It can be reused multiple times for building and running proof market
+binaries.
+
 ```
-docker build -t proof-market-toolchain-docker .  
-docker run -it --mount type=bind,src="$(PWD)",target=/proof-market-toolchain proof-market-toolchain-docker  sh ./build.sh
+docker build -t proof-market-toolchain .
+```
+
+Next, run a container and build the proof market in it.
```
Comment (arakelov) [inline README.md:101]: It is not a Proof market. It is the proof market toolchain

**Review by arakelov — APPROVED** [review]
It is necessary to update README more carefully

File: README.md (hunk 1 above)
Comment (NickVolynkin) [reply README.md]: fixed

**Pull request merged**